lib/block-supports/layout.php

```diff
@@ -887,6 +890,25 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 	return $processor->get_updated_html();
 }
 
+/**
+ * Add a `render_block_data` filter to fetch the parent block layout data.
+ */
+add_filter(
+	'render_block_data',
+	function ( $parsed_block, $source_block, $parent_block ) {
+		/**
+		 * Check if the parent block exists and if it has a layout attribute.
+		 * If it does, add the parent layout to the parsed block.
+		 */
+		if ( $parent_block && isset( $parent_block->parsed_block['attrs']['layout'] ) ) {
+			$parsed_block['parentLayout'] = $parent_block->parsed_block['attrs']['layout'];
```

Comment:
Editing the `$parsed_block` in a filter is messing with Interactivity API server side rendering.

Variables comparing function:
https://github.com/WordPress/wordpress-develop/blob/9a616a573434b432a5efd4de21039250658371fe/src/wp-includes/interactivity-api/interactivity-api.php#L49

Filter:
https://github.com/WordPress/wordpress-develop/blob/9a616a573434b432a5efd4de21039250658371fe/src/wp-includes/interactivity-api/interactivity-api.php#L69

I'm just writing this so I can follow it up on Monday, I need to work on a way to prevent SSR processing errors when someone filters a `parsed_block` (Thinking about giving last priority possible, as we compare in `render_block` filter, after all `render_block_data` happened. )

Reply:
For the record, we are comparing the parsed_block array ourselves because modifying it (adding a key like `$parsed_block['is_root_interactive_block'] = true`) broke the md5 comparison done in the block supports:

- Related PR: https://github.com/WordPress/gutenberg/pull/52557
- Block supports use of parsed_block md5: https://github.com/WordPress/wordpress-develop/blob/9a616a573434b432a5efd4de21039250658371fe/src/wp-includes/block-supports/elements.php#L19

Reply:
Thanks for the heads up! Yeah the issue with elements block supports already surfaced (see [comment above](https://github.com/WordPress/gutenberg/pull/59057#issuecomment-1972580083)) and was fixed in #59535. `render_block_data` is a publicly available filter which any extension could use to alter the content of `$parsed_block` so we can't assume it to be static. The fix will be added to core in 6.6; I guess the Interactivity API logic should also be updated accordingly.

Reply:
So, we will be able to set something like `$parsed_block['is_root_interactive_block'] = true` during `render_block_data`, and the check if it exists during `render_block`, right?

Reply:
Yeah, that should work fine now!